modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -1666,6 +1764,128 @@ private Predicate _getPredicate(
+		}
+
+		JSONArray sanitizedJSONArray = _jsonFactory.createJSONArray();
+
+		if (!JSONUtil.isJSONArray(json)) {
+			json = "[]";
+		}
+
+		JSONArray jsonArray = _jsonFactory.createJSONArray(json);
+
+		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
+			_cpDefinitionOptionRelLocalService.getCPDefinitionOptionRels(
+				cpInstance.getCPDefinitionId());
+
+		for (CPDefinitionOptionRel cpDefinitionOptionRel :
+				cpDefinitionOptionRels) {
+
+			JSONObject jsonObject = _getCPDefinitionOptionRelJSONObject(
+				cpDefinitionOptionRel, jsonArray);
+
+			if ((cpDefinitionOptionRel.isRequired() ||
+				 cpDefinitionOptionRel.isSkuContributor()) &&
+				JSONUtil.isEmpty(jsonObject)) {
+
+				throw new CPDefinitionOptionRelException();
+			}
+
+			JSONArray cpDefinitionOptionRelValueJSONArray =
+				_getCPDefinitionOptionRelValueJSONArray(
+					cpDefinitionOptionRel, jsonObject);
+
+			JSONObject sanitizedJSONObject = _jsonFactory.createJSONObject(
+			).put(
+				"key", cpDefinitionOptionRel.getKey()
+			).put(
+				"skuOptionKey", cpDefinitionOptionRel.getKey()
+			).put(
+				"skuOptionName", cpDefinitionOptionRel.getName(locale)
+			).put(
+				"value", cpDefinitionOptionRelValueJSONArray
+			);
+
+			String commerceOptionTypeKey =
+				cpDefinitionOptionRel.getCommerceOptionTypeKey();
+
+			if (ArrayUtil.contains(
+					CPConstants.PRODUCT_OPTION_SKU_CONTRIBUTOR_FIELD_TYPES,
+					commerceOptionTypeKey) &&
+				!JSONUtil.isEmpty(cpDefinitionOptionRelValueJSONArray)) {
+
+				sanitizedJSONObject.put(
+					"skuOptionValueKey",
+					cpDefinitionOptionRelValueJSONArray.getString(0));
+			}
+
+			if (ArrayUtil.contains(
+					CPConstants.PRODUCT_OPTION_MULTIPLE_VALUES_FIELD_TYPES,
+					commerceOptionTypeKey)) {
+
+				sanitizedJSONObject.put(
+					"skuOptionValueNames", cpDefinitionOptionRelValueJSONArray);
+			}
+
+			if (!cpDefinitionOptionRel.isPriceContributor() ||
+				JSONUtil.isEmpty(cpDefinitionOptionRelValueJSONArray)) {
+
+				sanitizedJSONArray.put(sanitizedJSONObject);
+
+				continue;
+			}
+
+			CPDefinitionOptionValueRel cpDefinitionOptionValueRel =
+				_cpDefinitionOptionValueRelLocalService.
+					fetchCPDefinitionOptionValueRel(
+						cpDefinitionOptionRel.getCPDefinitionOptionRelId(),
+						cpDefinitionOptionRelValueJSONArray.getString(0));
+
+			if (cpDefinitionOptionValueRel == null) {
+				sanitizedJSONArray.put(sanitizedJSONObject);
+
+				continue;
+			}
+
+			BigDecimal price = BigDecimal.ZERO;
+
+			if (cpDefinitionOptionRel.isPriceTypeStatic()) {
+				price = cpDefinitionOptionValueRel.getPrice();
+			}
+
+			sanitizedJSONObject.put(
+				"price", String.valueOf(price)
+			).put(
+				"priceType", cpDefinitionOptionRel.getPriceType()
+			);
+
+			CPInstance cpDefinitionOptionValueRelCPInstance =
+				cpDefinitionOptionValueRel.fetchCPInstance();
+
+			if (cpDefinitionOptionValueRelCPInstance != null) {
+				sanitizedJSONObject.put(
+					"quantity",
+					String.valueOf(cpDefinitionOptionValueRel.getQuantity())
```

Comment:
do we want it as string?

Reply:
Currently, we are storing it as a string, if we change it and in some systems there are open orders, then the JSON string won't match while adding a product to cart.